getting_started/scripting/c_sharp/c_sharp_style_guide.rst

```diff
@@ -0,0 +1,298 @@
+        {
+            Value,
+            AnotherValue                          // No blank line before '}'.
+        }
+                                                  // Blank line around inner types.
+        public const int SomeConstant = 1;
+        public const int AnotherConstant = 2;
+
+        private Vector3 _x;
+        private Vector3 _y;                       // Related constants or fields can be
+                                                  // grouped together.
+        private float _width;
+        private float _height;
+
+        public int MyProperty { get; set; }
+                                                  // Blank line around properties.
+        public void MyMethod()
+        {
+            // Some comment.
+            AnotherMethod();                      // No blank line after a comment.
+        }
+                                                  // Blank line around methods.
+        public void AnotherMethod()
+        {
+        }
+    }
+
+Consider breaking a line when it's longer than 120 characters. And it's also a good practice to
+insert a line feed(LF) character at the end of a file because some utilities have trouble
+recognizing the last line without it (i.e. Linux's *cat* command).
+
+Using Spaces
+------------
+
+Insert a space,
+
+* Around a binary and tertiary operator.
+* Between an opening parenthesis and *if*, *for*, *foreach*, *catch*, *while*, *lock* or *using* keywords.
+* Before and within a single line accessor block.
+* Between accessors in a single line accessor block.
+* After a comma.
+* After a semi-colon in a *for* statement.
+* After a colon in a single line *case* statement.
+* Around a colon in a type declaration.
+* Around a lambda arrow.
+* After a single line comment symbol('//'), and before it if used at the end of a line.
+
+Do not use a space,
+
+* After a type cast parentheses.
+* Within single line initializer braces.
+
+The following example shows a proper use of spaces, according to some of the the above mentioned conventions:
+
+.. code-block:: csharp
+
+    public class MyClass<A, B> : Parent<A, B>
+    {
+        public float MyProperty { get; set; }
+
+        public float AnotherProperty
+        {
+            get { return MyProperty; }
+        }
+
+        public void MyMethod()
+        {
+            int[] values = {1, 2, 3, 4}; // No space within initializer brackets.
+            int sum = 0;
+
+            // Single line comment.
+            for (int i = 0; i < values.Length; i++)
+            {
+                switch (i)
+                {
+                    case 3: return;
+                    default:
+                        sum += i > 2 ? 0 : 1;
+                        break;
+                }
+            }
+
+            i += (int)MyProperty; // No space after a type cast.
+        }
+    }
+
+Naming Conventions
+------------------
+
+Use upper camel case (or *'Pascal case'*) for all namespaces, type names and member level identifiers
```

Comment:
It's best when the name itself is visually descriptive. I suggest that we write it as `PascalCase`